sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/export/MultiMetricProducerReader.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.export;
+
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.CollectionRegistration;
+import io.opentelemetry.sdk.metrics.export.MetricProducer;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Helper class allowing {@link MetricReader} implementations to easily read from multiple {@link
+ * MetricProducer}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public final class MultiMetricProducerReader {
```

Comment:
When SdkMeterProvider starts, it calls each registered reader with an instance of `CollectionRegistration`. `CollectionRegistration` provides accessors to a list of `MetricProducers`, and each `MetricProducer` provides metrics via `Collection<MetricData> produce(Resource)`.

The `Resource` argument on `produce(..)` is necessary so SdkMeterProvider can inform the producer of the `Resource`, and the producer is expected to include that resource on all `MetricData`.

Most `MetricReader` will want to simply call something like `Collection<MetricData> readAll()`, and not care about the details of reading from each each reader. This helper class facilitates that. 

However, its designed so that a reader could potentially read only from some of the MetricProducers. Not sure this is necessary, and we could hide all of these details and expose a single `Collection<MetricData> readAll()` method from `CollectionRegistration` instead.

Reply:
> Not sure this is necessary, and we could hide all of these details and expose a single Collection<MetricData> readAll() method from CollectionRegistration instead.

Decided this was the way to go and pushed up a simplifying commit.